lxd/instance/drivers/driver_qemu.go

```diff
@@ -5855,11 +5865,71 @@ func (d *qemu) Info() instance.Info {
 		data.Version = "unknown" // Not necessarily an error that should prevent us using driver.
 	}
 
+	// Check IO-uring support.
+	supported, err := d.checkFeature(qemuPath, "-drive", "file=/dev/null,format=raw,aio=io_uring,file.locking=off")
+	if err != nil {
+		data.Error = fmt.Errorf("QEMU failed to run a feature check: %w", err)
+		return data
+	}
+
+	if supported {
+		data.Features = append(data.Features, "io_uring")
+	}
+
 	data.Error = nil
 
 	return data
 }
 
+func (d *qemu) checkFeature(qemu string, args ...string) (bool, error) {
+	pidFile, err := ioutil.TempFile("", "")
+	if err != nil {
+		return false, err
+	}
+	defer os.Remove(pidFile.Name())
+
+	qemuArgs := []string{
+		"qemu",
+		"-S",
+		"-nographic",
+		"-nodefaults",
+		"-daemonize",
```

Comment:
Is there a reason you used `-daemonize` here rather than just running the process and checking the exit code? Does it run indefinitely unless its killed?

Reply:
That's correct, it will keep running until killed.

The way it's done here, a failure (unsupported option) will cause `Wait()` to fail, on success, qemu will start and will run indefinitely needing to be killed.

Reply:
OK thanks